django/db/backends/sqlite3/_functions.py

```diff
@@ -0,0 +1,471 @@
+    create_deterministic_function('COS', 1, _sqlite_cos)
+    create_deterministic_function('COT', 1, _sqlite_cot)
+    create_deterministic_function('DEGREES', 1, _sqlite_degrees)
+    create_deterministic_function('EXP', 1, _sqlite_exp)
+    create_deterministic_function('FLOOR', 1, _sqlite_floor)
+    create_deterministic_function('LN', 1, _sqlite_ln)
+    create_deterministic_function('LOG', 2, _sqlite_log)
+    create_deterministic_function('LPAD', 3, _sqlite_lpad)
+    create_deterministic_function('MD5', 1, _sqlite_md5)
+    create_deterministic_function('MOD', 2, _sqlite_mod)
+    create_deterministic_function('PI', 0, lambda: pi)
+    create_deterministic_function('POWER', 2, _sqlite_power)
+    create_deterministic_function('RADIANS', 1, _sqlite_radians)
+    create_deterministic_function('REPEAT', 2, _sqlite_repeat)
+    create_deterministic_function('REVERSE', 1, _sqlite_reverse)
+    create_deterministic_function('RPAD', 3, _sqlite_rpad)
+    create_deterministic_function('SHA1', 1, _sqlite_sha1)
+    create_deterministic_function('SHA224', 1, _sqlite_sha224)
+    create_deterministic_function('SHA256', 1, _sqlite_sha256)
+    create_deterministic_function('SHA384', 1, _sqlite_sha384)
+    create_deterministic_function('SHA512', 1, _sqlite_sha512)
+    create_deterministic_function('SIGN', 1, _sqlite_sign)
+    create_deterministic_function('SIN', 1, _sqlite_sin)
+    create_deterministic_function('SQRT', 1, _sqlite_sqrt)
+    create_deterministic_function('TAN', 1, _sqlite_tan)
+    # Don't use the built-in RANDOM() function because it returns a value
+    # in the range [-1 * 2^63, 2^63 - 1] instead of [0, 1).
+    connection.create_function('RAND', 0, random.random)
+    connection.create_aggregate('STDDEV_POP', 1, StddevPop)
+    connection.create_aggregate('STDDEV_SAMP', 1, StddevSamp)
+    connection.create_aggregate('VAR_POP', 1, VarPop)
+    connection.create_aggregate('VAR_SAMP', 1, VarSamp)
+
+
+def _sqlite_datetime_parse(dt, tzname=None, conn_tzname=None):
+    if dt is None:
+        return None
+    try:
+        dt = typecast_timestamp(dt)
+    except (TypeError, ValueError):
+        return None
+    if conn_tzname:
+        dt = dt.replace(tzinfo=timezone_constructor(conn_tzname))
+    if tzname is not None and tzname != conn_tzname:
+        tzname, sign, offset = split_tzname_delta(tzname)
+        if offset:
+            hours, minutes = offset.split(':')
+            offset_delta = timedelta(hours=int(hours), minutes=int(minutes))
+            dt += offset_delta if sign == '+' else -offset_delta
+        dt = timezone.localtime(dt, timezone_constructor(tzname))
+    return dt
+
+
+def _sqlite_date_trunc(lookup_type, dt, tzname, conn_tzname):
+    dt = _sqlite_datetime_parse(dt, tzname, conn_tzname)
+    if dt is None:
+        return None
+    if lookup_type == 'year':
+        return "%i-01-01" % dt.year
+    elif lookup_type == 'quarter':
+        month_in_quarter = dt.month - (dt.month - 1) % 3
```

Comment:
This could be simplified to match the implementation in `_sqlite_datetime_extract()`:

```suggestion
        month_in_quarter = ceil(dt.month / 3)
```